web_external/js/views/widgets/WmsLayersListWidget.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+/**
+* This widget displays the list of WMS layers
+*/
+minerva.views.WmsLayersListWidget = minerva.View.extend({
+
+    events: {
+        'submit #m-add-layers-form': function (e) {
+            e.preventDefault();
+            var wmsSource = this.dataset;
+            var hostName = (wmsSource.get('meta').minerva.wms_params.base_url).slice(0, 28);
```

Comment:
Since we need the separate parts of the base url, go ahead and parse it out on the server side with the getUrlParts method I mentioned, then save the parts into the metadata of the dataset.  Then we should be able to access them here.